sentry_sdk/integrations/gql.py

```diff
@@ -0,0 +1,170 @@
+
+
+class GQLIntegration(Integration):
+    identifier = "gql"
+
+    @staticmethod
+    def setup_once() -> None:
+        if tuple(int(num) for num in gql.__version__.split(".")) < MIN_GQL_VERSION:
+            raise DidNotEnable(
+                "GQLIntegration is only supported for GQL versions %s and above."
+                % ".".join(str(num) for num in MIN_GQL_VERSION)
+            )
+        _patch_execute()
+
+
+def _data_from_document(document):
+    # type: (DocumentNode) -> EventDataType
+    try:
+        operation_ast = get_operation_ast(document)
+        data = {"query": print_ast(document)}  # type: EventDataType
+
+        if operation_ast is not None:
+            data["variables"] = operation_ast.variable_definitions
+            if operation_ast.name is not None:
+                data["operationName"] = operation_ast.name.value
+
+        return data
+    except (AttributeError, TypeError):
+        return dict()
+
+
+def _transport_method(transport):
+    # type: (Union[Transport, AsyncTransport]) -> str
+    """
+    The RequestsHTTPTransport allows defining the HTTP method; all
+    other transports use POST.
+    """
+    try:
+        return transport.method
+    except AttributeError:
+        return "POST"
+
+
+def _request_info_from_transport(transport):
+    # type: (Union[Transport, AsyncTransport, None]) -> Dict[str, str]
+    if transport is None:
+        return {}
+
+    request_info = {
+        "method": _transport_method(transport),
+    }
+
+    try:
+        request_info["url"] = transport.url
+    except AttributeError:
+        pass
+
+    return request_info
+
+
+def _patch_execute():
+    # type: () -> None
+    real_execute = gql.Client.execute
+
+    def sentry_patched_execute(self, document, *args, **kwargs):
+        # type: (gql.Client, DocumentNode, Any, Any) -> Any
+        hub = Hub.current
+        if hub.get_integration(GQLIntegration) is None:
+            return real_execute(self, document, *args, **kwargs)
+
+        with _graphql_event_processing(self, document):
+            try:
+                return real_execute(self, document, *args, **kwargs)
+            except TransportQueryError as e:
+                event, hint = event_from_exception(
+                    e,
+                    client_options=hub.client.options
+                    if hub.client is not None
+                    else None,
+                    mechanism={"type": "gql", "handled": False},
+                )
+
+                hub.capture_event(event, hint)
+                raise e
+
+    gql.Client.execute = sentry_patched_execute
+
+
+def _make_gql_event_processor(client, document):
+    # type: (gql.Client, DocumentNode) -> EventProcessor
+    def processor(event, hint):
+        # type: (Dict[str, Any], Dict[str, Any]) -> Dict[str, Any]
+        try:
+            errors = hint["exc_info"][1].errors
+        except (AttributeError, KeyError):
+            errors = None
+        _deep_update(
```

Comment:
An alternative to defining a new function would be utilizing [dict.setdefault](https://docs.python.org/3.11/library/stdtypes.html#dict.setdefault) which [we use in lots of places](https://github.com/search?q=repo%3Agetsentry%2Fsentry-python%20setdefault&type=code). But that's just a suggestion, this works too.

Reply:
I deleted this function per your previous comment